synapse/handlers/device.py

```diff
@@ -320,6 +324,8 @@ async def handle_room_un_partial_stated(self, room_id: str) -> None:
 
 
 class DeviceHandler(DeviceWorkerHandler):
```

Comment:
What's the difference between `DeviceHandler` and `DeviceWorkerHandler` now? Has this changed as a result of your patch here?

Reply:
I moved `get_dehydrated_device` from `DeviceHandler` to `DeviceWorkerHandler`. The only other change is that their `device_list_updater` are now `DeviceListUpdater` or `DeviceListWorkerUpdater`, respectively. The latter has an overridden method (`user_device_resync`) to call across replication to the main process.